bundles/org.eclipse.swt/Eclipse SWT Drag and Drop/cocoa/org/eclipse/swt/dnd/DropTarget.java

```diff
@@ -434,7 +442,20 @@ static long dropTargetProc(long id, long sel, long arg0) {
 	if (display == null || display.isDisposed()) return 0;
 	Widget widget = display.findWidget(id);
 	if (widget == null) return 0;
-	DropTarget dt = (DropTarget)widget.getData(DND.DROP_TARGET_KEY);
+	DropTarget dt = (DropTarget) widget.getData(DND.DROP_TARGET_KEY);
+	if (dt == null && widget instanceof Label) {
+		while (widget != null && !(widget instanceof Shell)) {
+		    dt = (DropTarget) widget.getData(DND.DROP_TARGET_KEY);
+		    if (dt != null) {
+		        break;
+		    }
+		    if (widget instanceof Control) {
+		        Composite widgetParent = ((Control) widget).getParent();
+		        widget = widgetParent;
+		    }
+		    else break;
+		    }
```

Comment:
That looks more logical to me.  

A nit.  The indentation of line 457 one tab too many.

Reply:
I've now correctly indented the curly braces.